sql/catalyst/src/main/scala/org/apache/spark/sql/sources/filters.scala

```diff
@@ -36,10 +36,18 @@ abstract class Filter {
    */
   def references: Array[String]
 
+  def sql: String
+
   protected def findReferences(value: Any): Array[String] = value match {
     case f: Filter => f.references
     case _ => Array.empty
   }
+
+  protected def quoteIdentifier(name: String): String = {
```

Comment:
Why not use `CatalogV2Implicits` to get the `quoted` method?

Reply:
This code is borrowed from `org.apache.spark.sql.catalyst.util.quoteIdentifier` which is a package util, while `CatalogV2Implicits.quoted` is not a public util function. We'd better unify the two, I think.
However, this code is introduced by the needs in the delete test case. Now the test code is updated according to your suggestion below, which left this function (sources.filter.sql) unused. I have removed this function in the latest code. If we need this function in future (like translating filters to sql string in jdbc), we then submit a new pr.